internal/pkg/api/handleStatus.go

```diff
@@ -61,17 +61,17 @@ func (st StatusT) authenticate(r *http.Request) (*apikey.APIKey, error) {
 	return authAPIKey(r, st.bulk, st.cache)
 }
 
-func (st StatusT) handleStatus(_ *zerolog.Logger, r *http.Request, rt *Router) (resp StatusResponse, status proto.StateObserved_Status) {
+func (st StatusT) handleStatus(_ *zerolog.Logger, r *http.Request, rt *Router) (resp StatusResponse, state client.UnitState) {
 	authed := true
 	if _, aerr := st.authfn(r); aerr != nil {
 		log.Debug().Err(aerr).Msg("unauthenticated status request, return short status response only")
 		authed = false
 	}
 
-	status = rt.sm.Status()
+	state = rt.sm.State()
```

Comment:
It seems that changing the type of the returned status also changed its string representation. Was this expected?
This affected `elastic-package stack` healthchecks (see https://github.com/elastic/elastic-package/pull/1042). I wonder if this can affect other things.

Reply:
I don't think this was intentional, it affected APM server's tests in the same way.

I filed it as a bug in https://github.com/elastic/elastic-agent/issues/1762